app/models/course.rb

```diff
@@ -811,8 +810,27 @@ def find_a_level_subject_requirement!(uuid)
     requirement.with_indifferent_access
   end
 
+  def ensure_site_statuses_match_full_time
+    site_statuses.each do |site_status|
+      update_vac_status('full_time', site_status)
+    end
+  end
+
   private
 
+  def update_vac_status(study_mode, site_status)
+    case study_mode
+    when 'full_time'
+      site_status.update(vac_status: :full_time_vacancies)
+    when 'part_time'
+      site_status.update(vac_status: :part_time_vacancies)
+    when 'full_time_or_part_time'
+      site_status.update(vac_status: :both_full_time_and_part_time_vacancies)
+    else
+      raise "Unexpected study mode #{study_mode}"
+    end
+  end
```

Comment:
This method was in a controller concern and was included in the Course model (only on course) so I remove the module and move the code here.